twitch_hdt_ebs/views.py

```diff
@@ -277,6 +283,72 @@ def put(self, request, format=None):
 		return Response(serializer.validated_data)
 
 
+class ActiveChannelsView(APIView):
+	permission_classes = (HasApiSecretKey, )
+
+	ALPHABET = string.ascii_letters + string.digits
+
+	def generate_digest_from_deck_list(self, id_list: List[str]) -> str:
+		sorted_cards = sorted(id_list)
+		m = hashlib.md5()
+		m.update(",".join(sorted_cards).encode("utf-8"))
+		return m.hexdigest()
+
+	def get_shortid_from_digest(self, digest: str) -> str:
+		return int_to_string(int(digest, 16), ActiveChannelsView.ALPHABET)
+
+	def get_shortid_from_deck_list(self, cards: List[List[int]]) -> str:
+		card_list = []
+		for dbf_id, count, _ in cards:
+			card = Card.objects.get(dbf_id=int(dbf_id))
+			card_list.extend([card.card_id for i in range(count)])
+		digest = self.generate_digest_from_deck_list(card_list)
+		return self.get_shortid_from_digest(digest)
+
+	def to_deck_url(self, deck) -> str:
+		deck_cards_count = []
+		for dbf_id, count, _ in deck.get("cards", []):
+			deck_cards_count.append(f"{dbf_id}_{count}")
+		deck_key = ",".join(sorted(deck_cards_count))
+
+		deck_url = caches["default"].get(deck_key)
+		if not deck_url:
+			short_id = self.get_shortid_from_deck_list(deck.get("cards", []))
+			deck_url = f"https://hsreplay.net/decks/{short_id}/"
+			caches["default"].set(deck_key, deck_url, timeout=1200)
```

Comment:
Let's only cache the short_id instead of the URL, so we can do clever things like utms and extra params.